server/rest/src/main/java/org/infinispan/rest/server/InfinispanResponse.java

```diff
@@ -0,0 +1,153 @@
+package org.infinispan.rest.server;
+
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.Date;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+import org.infinispan.rest.server.operations.mediatypes.MediaType;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufUtil;
+import io.netty.buffer.Unpooled;
+import io.netty.handler.codec.http.DefaultFullHttpResponse;
+import io.netty.handler.codec.http.FullHttpResponse;
+import io.netty.handler.codec.http.HttpHeaderNames;
+import io.netty.handler.codec.http.HttpHeaderValues;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.netty.handler.codec.http.HttpUtil;
+import io.netty.handler.codec.http.HttpVersion;
+import io.netty.handler.codec.http2.HttpConversionUtil;
+
+public class InfinispanResponse {
```

Comment:
@slaskawi You know as well as I do that we only notice these performance problems right before the release :)

Speaking of performance, you mentioned in a comment that the new version is slightly slower in JMH, but I didn't see any follow-up on what was causing the slowdown.